CMakeLists.txt

```diff
@@ -81,7 +81,12 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   set (WARN_FLAGS "${WARN_FLAGS} -Wno-covered-switch-default")
   set (WARN_FLAGS "${WARN_FLAGS} -Wno-missing-noreturn -Wno-unknown-pragmas")
   set (WARN_FLAGS "${WARN_FLAGS} -Wno-gnu-zero-variadic-macro-arguments")
-  set (WARN_FLAGS "${WARN_FLAGS} -Wconversion -Werror")
+  set (WARN_FLAGS "${WARN_FLAGS} -Wconversion")
+  if ((NOT CMAKE_CXX_COMPILER_VERSION STREQUAL "") AND
+      CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "11.0")
```

Comment:
does Clang have a 11.0 release?

Reply:
Ok, the problem was with the MacOS Clang, which reports itself as:

omalley@dhcp-10-16-1-45> clang --version
Apple clang version 11.0.0 (clang-1100.0.33.8)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

I see your point that there isn't an upstream clang 11. I'm not sure what Apple is doing with their version numbers or the right way to detect it.

Reply:
Yeah, I think Apple uses the XCode version.
Instead of `CMAKE_CXX_COMPILER_VERSION` should we use
`
"${COMPILER_VERSION_FULL}" MATCHES ".*clang-11"
`

Reply:
Source: https://github.com/apache/arrow/blob/master/cpp/cmake_modules/CompilerInfo.cmake#L54

Reply:
I used CMAKE_HOST_APPLE and the CMAKE_CXX_COMPILER_VERSION > 11. No doubt when some other OS is using clang 8?, we'll need to generalize it.

Reply:
On MacOS Mojave (10.14), COMPILER_VERSION_FULL is undefined on cmake...

Reply:
Ah, I see now that they are using the compiler itself to get that value. Current XCode doesn't match their patterns. :(